core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -498,6 +498,67 @@ def fit(self, **kwargs):
             logger.warning(f'\tWarning: Model has no time left to train, skipping model... (Time Left = {round(kwargs["time_limit"], 1)}s)')
             raise TimeLimitExceeded
 
+    def fit_with_prune(self, max_num_fit=3, prune_threshold=0., num_shuffle_sets=50, **kwargs):
```

Comment:
Add early_stopping_rounds, currently the functionality acts like `early_stopping_rounds=1`, but likely better results will come from `early_stopping_rounds=2`

Reply:
Add TODO: Consider returning list of models, similar to HPO, as it is theoretically possible that ensembling the different iterations could result in a better model.

We don't need to do this right away, as it is not necessary to get the primary benefit. It will however help greatly in debugging and testing all of the iterations against test data rather than val data to see if we are overfitting in this method.

Reply:
Adding early stopping with rounds >= 1 would mean we would have to adjust the threshold too, right? Otherwise if feature pruning sucked once we shouldn't take away even more features from that point.

Reply:
That is an implementation detail, in general we want to be adjusting the threshold but that is a separate piece of the overall functionality.